tests/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_4_durability_and_cover/test_table_4_3.py

```diff
@@ -0,0 +1,179 @@
+"""Test for the concrete structural class
+according to Table 4.3 from NEN-EN 1992-1-1+C2:2011: Chapter 4 - Durability and cover to reinforcement.
+"""
+
+import pytest
+
+from blueprints.codes.eurocode.nen_en_1992_1_1_c2_2011.chapter_4_durability_and_cover.table_4_1 import (
+    Carbonation,
+    Chemical,
+    Chloride,
+    ChlorideSeawater,
+    ExposureClasses,
+    FreezeThaw,
+)
+from blueprints.codes.eurocode.nen_en_1992_1_1_c2_2011.chapter_4_durability_and_cover.table_4_3 import (
+    ConcreteMaterial,
+    ConcreteStrengthClass,
+    ConcreteStructuralClass,
+    ConcreteStructuralClassBase,
+    ConcreteStructuralClassCalculator,
+)
+from blueprints.type_alias import YEARS
+
+DUMMY_EXPOSURE_CLASSES = ExposureClasses(
+    carbonation=Carbonation.XC2,
+    chloride=Chloride.XD1,
+    chloride_seawater=ChlorideSeawater.XS1,
+    freeze=FreezeThaw.NA,
+    chemical=Chemical.XA2,
+)
+
+
+class TestConcreteStructuralClassCalculator:
+    """Unit tests for the ConcreteStructuralClassCalculator class."""
+
+    @pytest.mark.parametrize(
+        ("design_working_life", "expected_structural_class", "expected_explanation"),
+        [(50, 4, "(50 years)"), (75, 5, "(75 years)"), (100, 6, "(100 years)")],
+    )
+    def test_structural_class_delta_design_working_life(
+        self, design_working_life: YEARS, expected_structural_class: int, expected_explanation: str
+    ) -> None:
+        """Test the _structural_class_delta_design_working_life method of the ConcreteStructuralClassCalculator class."""
+        calculator = ConcreteStructuralClassCalculator(
+            DUMMY_EXPOSURE_CLASSES,
+            design_working_life,
+            ConcreteMaterial(ConcreteStrengthClass("C20/25")),
+            False,
+            False,
+        )
+
+        calculator._structural_class_delta_design_working_life()  # noqa: SLF001
+        calculator._calculated = True  # noqa: SLF001
+
+        assert calculator.structural_class == expected_structural_class
+        assert calculator.explanation.endswith(expected_explanation)
+
+    @pytest.mark.parametrize(
+        ("plate_geometry", "expected_structural_class", "expected_explanation"),
+        [(True, 3, "(plate geometry)"), (False, 4, "(no plate geometry)")],
+    )
+    def test_structural_class_delta_plate_geometry(self, plate_geometry: bool, expected_structural_class: int, expected_explanation: str) -> None:
+        """Test the _structural_class_delta_plate_geometry method of the ConcreteStructuralClassCalculator class."""
+        calculator = ConcreteStructuralClassCalculator(
+            DUMMY_EXPOSURE_CLASSES,
+            50,
+            ConcreteMaterial(ConcreteStrengthClass("C20/25")),
+            plate_geometry,
+            False,
+        )
+
+        calculator._structural_class_delta_plate_geometry()  # noqa: SLF001
+        calculator._calculated = True  # noqa: SLF001
```

Comment:
This feels like a workaround. The calculator shouldn't be dependent of a private interface to get the final calculations.

Reply:
changed it.